### PR TITLE
[grafana] add extraMounts to plugins sidecar

### LINKED ISSUE #3351: [Grafana] Adding extraVolumeMount to sidecar to run custom script
Problem:
The grafana chart has configuration to have sidecars and even a `script` env variable to set the path to a custom script. For the moment, there is no `extraVolumes` and `extraVolumeMounts` to use to mount a script in the sidecar container. The only additional volumes that can be add are for the main Grafana container. Maybe is there another way to run a custom script that I am not aware of. The only other way I thought of is to modify the sidecar image to already have my script but that seems like more work than needed.

Objective:
I would like to mount a custom script to mount to a sidecar and run it everytime plugins are reloaded like described [here](https://github.com/grafana/helm-charts/blob/350783e92680eb98ff0c2b75eb0502a2dcc79fcb/charts/grafana/values.yaml#L1053).

Solution:
Add `extraVolumes` and `extraVolumeMounts` to the sidecar options.

### PR DESCRIPTION
**Pull request by coutug**
Fixes #3351

### CODE REVIEW

File: charts/grafana/templates/_pod.tpl [hunk 1]
```diff
@@ -857,6 +857,9 @@ containers:
     volumeMounts:
       - name: sc-plugins-volume
         mountPath: "/etc/grafana/provisioning/plugins"
+      {{- with .Values.sidecar.plugins.extraMounts }}
+      {{- toYaml . | trim | nindent 6 }}
```
Comment (jkroepke) [inline charts/grafana/templates/_pod.tpl:861]: why trim?

**Review by jkroepke — COMMENTED** [review]
Please bump the chart version, thanks!

File: charts/grafana/templates/_pod.tpl (hunk 1 above)
Comment (coutug) [reply charts/grafana/templates/_pod.tpl]: not really required, I just left it the same as other extraMounts (like the sc-alerts one)

**Review by jkroepke — APPROVED** [review]

File: charts/grafana/templates/_pod.tpl [hunk 2]
```diff
@@ -857,6 +857,9 @@ containers:
     volumeMounts:
       - name: sc-plugins-volume
         mountPath: "/etc/grafana/provisioning/plugins"
+      {{- with .Values.sidecar.plugins.extraMounts }}
```
Comment (jkroepke) [inline charts/grafana/templates/_pod.tpl:860]: please add this to all sidecars

Comment (coutug) [reply charts/grafana/templates/_pod.tpl:860]: Sure, do you want me to remove the trim function from existing extraMount or add it to the new extraMount for the sake of consistency?

Comment (jkroepke) [reply charts/grafana/templates/_pod.tpl:860]: Yes, add it for the sake of consistency.

**Review by jkroepke — APPROVED** [review]

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**